server/embedded/src/org/labkey/embedded/LabKeyServer.java

```diff
@@ -283,6 +326,61 @@ private String getPropValue(Map<Integer, String> propValues, Integer resourceKey
                 return val != null && !val.isBlank() ? val.trim() : defaultValue;
             }
 
+            private ContextResource getJmsResource()
+            {
+                JmsProperties jmsProps = jmsSource();
+                if (jmsProps.getBrokerURL() == null)
+                {
+                    return null;
+                }
+
+                ContextResource jmsResource = new ContextResource();
+                jmsResource.setName("jms/ConnectionFactory");
+                jmsResource.setAuth("Container");
+                jmsResource.setType(jmsProps.getType());
+                jmsResource.setProperty("factory", jmsProps.getFactory());
+                jmsResource.setProperty("description", jmsProps.getDescription());
+                jmsResource.setProperty("brokerURL", jmsProps.getBrokerURL());
+                jmsResource.setProperty("brokerName", jmsProps.getBrokerName());
+                return jmsResource;
+            }
+            private ContextResource getLdapResource()
+            {
+                LdapProperties ldapProps = ldapSource();
+                if (ldapProps.getHost() == null)
+                {
+                    return null;
+                }
+
+                ContextResource ldapResource = new ContextResource();
+                ldapResource.setName("ldap/ConfigFactory");
+                ldapResource.setAuth("Container");
+                ldapResource.setType(ldapProps.getType());
+                ldapResource.setProperty("host", ldapProps.getHost());
+                ldapResource.setProperty("port", Integer.toString(ldapProps.getPort()));
+                if (ldapProps.getPrincipal() != null)
+                {
+                    ldapResource.setProperty("principal", ldapProps.getPrincipal());
+                }
+                if (ldapProps.getCredentials() != null)
+                {
+                    ldapResource.setProperty("credentials", ldapProps.getCredentials());
+                }
+                if (ldapProps.isUseSsl())
```

Comment:
Seems like this if is not needed since we can just as well set the property to false. (Same for `isUseTls` below.)

Reply:
True. I was aiming for to create the resource most similarly to how it would have been configured in the XML world, but this simplification should be safe.